Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c

```diff
@@ -269,6 +283,779 @@ VerifyValidVarWriteReadEnumDel (
+    IN CHAR16   *VariableName,
+    IN EFI_GUID *VendorGuid,
+    IN UINT32   BaseAttributes,
+    IN UINT32   ExtraSetAttributes
+  )
+{
+  UINTN DataSize = 0;
+  UINTN ActualDataSize = 0;
+  UINTN ActualVarAttributes;
+
+  VERIFY_SUCCEEDED (gRT->SetVariable (
+                            VariableName,
+                            VendorGuid,
+                            BaseAttributes | ExtraSetAttributes,
+                            DataSize,
+                            NULL),
+                    "Deleting %s",
+                    VariableName
+                    );
+
+  VERIFY_ARE_EQUAL (EFI_STATUS,
+                    EFI_NOT_FOUND,
+                    gRT->GetVariable (
+                          VariableName,
+                          VendorGuid,
+                          &ActualVarAttributes,
+                          &ActualDataSize,
+                          NULL),
+                    "Checking %s was deleted",
+                    VariableName
+                    );
+
+  VerifyVariableEnumerable (VariableName,
+                            VendorGuid,
+                            FALSE
+                            );
+}
+
+/**
+  Writes and updates a set of persistant variables in an interleaved order. The
+  order is intended to exercise the TA based AuthVar's memory management.
+
+  NOTE: The authvar TA v1.1 no longer implements block based memory, but this is
+  still and interesting test which exercises multiple operations.
+
+  This test will also verify the presence and accuracy of these variables from
+  a previous run. If the values do not match the PersistanceTestFlag will be cleared
+  and a subsequent run will reset the values.
+**/
+VOID
+VerifyAppendShrinkReplacePersist (
+    VOID
+  )
+{
+  STATIC CHAR16 PersistanceTestFlag[] = L"PersistanceTestFlag";
+  STATIC CHAR16 TestVariableName1[] = L"InterleaveTestVar1";
+  STATIC CHAR16 TestVariableName2[] = L"InterleaveTestVar2";
+  STATIC CHAR16 TestVariableName3[] = L"InterleaveTestVar3";
+  STATIC CONST UINT32 ChunkSize = 0x10;
+  STATIC CONST UINT32 MaxChunksPerVar = 10;
+  STATIC CONST UINT32 BufferSize = ChunkSize * MaxChunksPerVar * 3;
+  VOID *Buffer1;
+  VOID *Buffer2;
+  VOID *Buffer3;
+
+  UINTN Idx;
+  UINT32 ActualVarAttributes;
+  UINTN ActualDataSize;
+  UINT8 Persistanceflag;
+
+  STATIC CONST UINT32 BaseAttributes = (EFI_VARIABLE_NON_VOLATILE |
+                                        EFI_VARIABLE_RUNTIME_ACCESS |
+                                        EFI_VARIABLE_BOOTSERVICE_ACCESS);
+  EFI_STATUS Status;
+
+  mImageData = AllocateZeroPool (BufferSize);
+  VERIFY_IS_NOT_NULL (mImageData);
+
+  Buffer1 = mImageData;
+  Buffer2 = Buffer1 + ChunkSize * MaxChunksPerVar;
+  Buffer3 = Buffer2 + ChunkSize * MaxChunksPerVar;
+
+  mActualImageData = AllocatePool(BufferSize);
+  VERIFY_IS_NOT_NULL(mActualImageData);
+
+  // To allow for checking persistance it is important that the data
+  // is deterministic.
+  for(Idx = 0; Idx < BufferSize; Idx++) {

```

Comment:
Nit - Space after for
```suggestion
  for (Idx = 0; Idx < BufferSize; Idx++) {
```